docs/src/modules/spring/pages/event-sourced-entities.adoc

```diff
@@ -0,0 +1,160 @@
+= Implementing Event Sourced Entities with Spring
+:page-aliases: spring:eventsourced.adoc
+:page-supergroup-java-scala: Language
+:sample-url: https://github.com/lightbend/kalix-jvm-sdk/tree/main/samples/spring-eventsourced-shopping-cart
+
+include::ROOT:partial$include.adoc[]
+include::ROOT:partial$eventsourced.adoc[]
+
+https://docs.kalix.io/reference/glossary.html#event_sourced_entity[Event Sourced Entities] persist changes as events and snapshots. Kalix needs to serialize that data to send it to the underlying data store. However, we recommend that you do not persist your service's public API messages. This may introduce some overhead to convert from one type to an internal one but allows the service public interface logic to evolve independently of the data storage format, which should be private.
+
+The steps necessary to implement an Event Sourced Entity include:
+
+. Model the entity's state and its domain events.
+. Implementing behavior in command and event handlers.
+. Creating and initializing the Entity.
+
+
+The following sections walk through these steps using a shopping cart service as an example (working sample available {sample-url}[here {tab-icon}, window="new"]).
+
+== Model the entity
+
+Through our "Shopping Cart" Event Sourced Entity we expect to manage our cart, adding and removing items as we please. Being event-sourced means it will represent changes to state as a series of domain events. So let's have a look at what kind of model we expect to store and the events our entity might generate.
+
+[source,java]
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCart.java[tag=domain]
+----
+<1> Our `ShoppingCart` is fairly simple, being composed only by a `cartId` and a list of line items.
+<2> A `LineItem` represents a single product and the quantity we intend to buy.
+
+NOTE: Above we are taking advantage of the Java `record` to reduce the amount of boilerplate code, but you can use regular classes so long as they can be serialized.
```

Comment:
Maybe better to be explicit that it needs to be serializable to json and therefore properly annotated with Jackson annotations.

Reply:
Thanks, good to clarify it is "to json". I wonder if we should explicitely mention Jackson though, since although not the most likely, one could still decide to use another lib (e.g. gson) instead. But maybe I can mention Jackson as an example rather giving the impression it is required to use that one. 👍